src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+        private UsernameGenerationOptions _optionsCache;
+
+        public UsernameGenerationService(
+            ICryptoService cryptoService,
+            IApiService apiService,
+            IStateService stateService
+            )
+        {
+            _cryptoService = cryptoService;
+            _apiService = apiService;
+            _stateService = stateService;
+        }
+
+        public async Task<string> GenerateUsernameAsync(UsernameGenerationOptions options)
+        {
+            switch (options.Type)
+            {
+                case UsernameType.PlusAddressedEmail:
+                    return await GeneratePlusAddressedEmailAsync(options);
+                case UsernameType.CatchAllEmail:
+                    return await GenerateCatchAllAsync(options);
+                case UsernameType.ForwardedEmailAlias:
+                    return await GenerateForwardedEmailAliasAsync(options);
+                case UsernameType.RandomWord:
+                    return await GenerateRandomWordAsync(options);
+                default:
+                    return string.Empty;
+            }
+        }
+
+        public async Task<UsernameGenerationOptions> GetOptionsAsync()
+        {
+            if (_optionsCache == null)
+            {
+                var options = await _stateService.GetUsernameGenerationOptionsAsync();
+                if (options == null)
+                {
+                    _optionsCache = _defaultOptions;
+                }
+                else
+                {
+                    options.Merge(_defaultOptions);
+                    _optionsCache = options;
+                }
+            }
+
+            return _optionsCache;
+        }
+        public async Task SaveOptionsAsync(UsernameGenerationOptions options)
+        {
+            await _stateService.SetUsernameGenerationOptionsAsync(options);
+            _optionsCache = options;
+        }
+
+        public void ClearCache()
+        {
+            _optionsCache = null;
+        }
+
+        private async Task<string> GenerateRandomWordAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string randomWord = null;
+            if (options.RandomWordUsernameCapitalize == null)
+            {
+                options.RandomWordUsernameCapitalize = false;
+            }
+            if (options.RandomWordUsernameIncludeNumber == null)
+            {
+                options.RandomWordUsernameIncludeNumber = false;
+            }
+            var listLength = EEFLongWordList.Instance.List.Count - 1;
+            var wordIndex = await _cryptoService.RandomNumberAsync(0, listLength);
+            if (options.RandomWordUsernameCapitalize.GetValueOrDefault())
+            {
+                randomWord = Capitalize(EEFLongWordList.Instance.List[wordIndex]);
+            }
+            else
+            {
+                randomWord = EEFLongWordList.Instance.List[wordIndex];
+            }
```

Comment:
to avoid the `else` (and also the null assignment above) you can declare and assign directly `randomWord` with the word and then if necessary you make it upper case:
```c#
var randomWord = EEFLongWordList.Instance.List[wordIndex];
if (options.RandomWordUsernameCapitalize.GetValueOrDefault())
{
    randomWord = Capitalize(randomWord);
}
```